### PR TITLE
Added 128bit hash.

### PR DESCRIPTION
**Pull request by swapniel99**
Hi,

I have added all the functions available in mmh3.
I have also updated the test file generation code and added testcases.

Please review and pull.

Regards.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,17 +1,21 @@
...
 ## About:
 This is a pure python implementation of the murmur3 hash algorithm <https://code.google.com/p/smhasher/wiki/MurmurHash3>
 written for the times when you do not want to muck about building and installing c-modules etc.
 
-As this is pure python performance is FAR from optimal and if that is required a real c-module such as the
-one found here at <https://pypi.python.org/pypi/mmh3/2.0> is absolutely suggested!
+As this is pure python performance is FAR from optimal in CPython and if that is required a real c-module such as the
+one found here at <https://pypi.python.org/pypi/mmh3/2.3.1> is absolutely suggested!
+
+However this module performs extremely well in case of pypy(<http://pypy.org>), an alternative JIT version of python.
```
Comment (wc-duck) [inline README.md]: I'm a bit skeptical to this statement, you told me that there was a x20 improvement over the c-module in pypy? That is really alot?
How was the c-module built, was it built without optimizations? It just sounds a bit to good to be true ;)

File: README.md [hunk 2]
```diff
@@ -1,17 +1,21 @@
 # PYMMH3 - a pure python MurmurHash3 implementation.
 
-[![Build Status](https://travis-ci.org/wc-duck/pymmh3.svg?branch=master)](https://travis-ci.org/wc-duck/pymmh3)
+[![Build Status](https://travis-ci.org/swapniel99/pymmh3.svg?branch=master)](https://travis-ci.org/swapniel99/pymmh3)
```
Comment (wc-duck) [inline README.md]: This would need to be reverted before I merge the PR

File: README.md [hunk 3]
```diff
@@ -1,17 +1,21 @@
...
-one found here at <https://pypi.python.org/pypi/mmh3/2.0> is absolutely suggested!
+As this is pure python performance is FAR from optimal in CPython and if that is required a real c-module such as the
+one found here at <https://pypi.python.org/pypi/mmh3/2.3.1> is absolutely suggested!
+
+However this module performs extremely well in case of pypy(<http://pypy.org>), an alternative JIT version of python.
 
 ## Performance:
 Bad... we need not say more ;)
 
+But with pypy it is blazing fast compared to mmh3 :D
```
Comment (wc-duck) [inline README.md]: blazing compared to what ;)

File: pymmh3.py [hunk 4]
```diff
@@ -14,18 +14,17 @@
...
 This module is written to have the same format as mmh3 python package found here for simple conversions:
 
-https://pypi.python.org/pypi/mmh3/2.0 
+https://pypi.python.org/pypi/mmh3/2.3.1
 '''
 
-import sys
-
-if (sys.version_info > (3, 0)):
+import sys as _sys
```
Comment (wc-duck) [inline pymmh3.py]: why the deletion of sys, why not just keep it as import sys as it was?

File: pymmh3.py [hunk 5]
```diff
@@ -73,14 +72,345 @@ def fmix( h ):
...
         k1  = ( k1 * c1 ) & 0xFFFFFFFF
-        k1  = ( k1 << 15 | k1 >> 17 ) & 0xFFFFFFFF # _ROTL32
+        k1  = ( k1 << 15 | k1 >> 17 ) & 0xFFFFFFFF # inlined ROTL32
         k1  = ( k1 * c2 ) & 0xFFFFFFFF
         h1 ^= k1
 
+    #finalization
     return fmix( h1 ^ length )
 
+def hash128( key, seed = 0x0, x64arch = True ):
```
Comment (wc-duck) [inline pymmh3.py]: I really can't see why you would want the x64arch flag? That will force your users to check the current arch. Is there any performance-reason to have both implementations?
If there is a performance difference for this code, why not just do a selection of the function to use at module load and then always use the correct one for the current platform?

File: test/MurmurHash3.inc [hunk 6]
```diff
@@ -16,20 +16,25 @@
```
Comment (wc-duck) [inline test/MurmurHash3.inc]: I guess that all these changes is just an update from the smhasher repo?

File: test/unitest_solution_gen.cpp [hunk 7]
```diff
@@ -18,23 +18,44 @@ namespace mmh3 {
...
 int main( int argc, const char** argv )
 {
-	FILE* f   = fopen( "test/pg1260.txt", "rb" );
-	FILE* out1 = fopen( "test/solution_hash32_seed0.txt", "wb" );
-	FILE* out2 = fopen( "test/solution_hash32_seed1234ABCD.txt", "wb" );
-	char buffer[2048];
+    FILE* f    = fopen( "test/pg1260.txt", "rb" );
+    FILE* out1 = fopen( "test/solution_hash32_seed0.txt", "wb" );
+    FILE* out2 = fopen( "test/solution_hash32_seed1234ABCD.txt", "wb" );
+    FILE* out3 = fopen( "test/solution_hash128_x86_seed0.txt", "wb" );
```
Comment (wc-duck) [inline test/unitest_solution_gen.cpp]: is there really a reason to have solution files both for the x86 and the x64 implementations... if these aren't the same, isn't murmurhash broken?
I.e. it should be enough to have only one solution_hash128_***.txt

Comment (swapniel99) [inline test/unitest_solution_gen.cpp]: x86 and x64 implementations give different hash values by design.
You can verify this using mmh3, the c++ implementation.

File: test/MurmurHash3.inc (hunk 6 above)
Comment (swapniel99) [inline test/MurmurHash3.inc]: Yes.

File: pymmh3.py (hunk 5 above)
Comment (swapniel99) [inline pymmh3.py]: This is a feature in mmh3.
x86 on c++ module is optimized for 32 bit platforms. So if someone uses pymmh3, he will need x86 version to generate the same hash values as before. By default x64 will be used.
So I used it here to make it seamless an transparent. Please refer:
https://github.com/hajimes/mmh3

File: pymmh3.py (hunk 4 above)
Comment (swapniel99) [inline pymmh3.py]: That one is on me.. if we don't delete it, we can import sys from pymmh3.
I just could ignore it. My  i guess :P

File: README.md (hunk 3 above)
Comment (swapniel99) [inline README.md]: Blazing fast compared to mmh3: https://github.com/hajimes/mmh3
I have tested this ONLY on pypy. One of the example testing code was as follows:

#!/usr/bin/pypy
from mmh3 import hash
for i in range(1000000):
    hash('foo')

If you test with any other input, make sure entire input is read before 'for' loop begins. Else file reading diminishes performance.

File: README.md (hunk 1 above)
Comment (swapniel99) [inline README.md]: C module(https://github.com/hajimes/mmh3) performs extremely well in cpython.
But in pypy it is the other way round.

File: README.md (hunk 2 above)
Comment (swapniel99) [inline README.md]: Please could you do it?
This is the first time I used Travis CI :D

File: README.md (hunk 1 above)
Comment (wc-duck) [inline README.md]: What I ask is, how was the c-module built? I see that the JIT in pypy could work well with this code but I have a hard time to see that there should be x20 overhead in calling into a .pyd (dll), reading i string ( that is basically just passing a pointer ), and returning a 32-bit value.
I might be wrong but for me it sounds like something is off.
Was it you who built mmh3? In that case, how?
I might be wrong, I have been before ;) I just don't want to claim things that might not be true.

File: README.md (hunk 2 above)
Comment (wc-duck) [inline README.md]: Sure

File: pymmh3.py (hunk 4 above)
Comment (wc-duck) [inline pymmh3.py]: Is that usually something python-folks care about? :) Personally I just find it to clutter the code for no real gain.

File: pymmh3.py (hunk 5 above)
Comment (wc-duck) [inline pymmh3.py]: I saw that now as well, I agree that conformance is more important that anything else. My bad.

File: test/unitest_solution_gen.cpp (hunk 7 above)
Comment (wc-duck) [inline test/unitest_solution_gen.cpp]: Hmm... there you go! In my naivety I just thought them to be different versions of the same code optimized for different arch:es. I was again wrong... carry on ;)

**Comment by wc-duck** [comment]
By the way, I see no reason not to merge this. Great work, hopefully I get by it tomorrow.

File: README.md (hunk 1 above)
Comment (swapniel99) [inline README.md]: I installed mmh3 simply using pip

**Comment by swapniel99** [comment]
Hi, any update?

**Comment by wc-duck** [comment]
Sorry... personal-life got in the way. Merged now.

**Pull request merged**